src/BaseRequester.ts

```diff
@@ -0,0 +1,539 @@
+import axiosCreate from './axiosCreate'
+import {isBrowser, requiredArg} from './helpers'
+import defaultConfig from './defaultConfig'
+import {GRANT_TYPES, DEVICE_ID, IDEMPOTENT_HTTP_VERBS, MAX_TOKEN_LATENCY} from './constants'
+import {rateLimitWarning, RateLimitError} from './errors'
+import type {AxiosRequestConfig, AxiosResponse, AxiosError} from './axiosCreate'
+import type {CredentialsResponse} from './interfaces'
+
+
+interface Common {
+  redirect_uri?: string
+  user_agent?: string
+  device_id?: string
+  grant_type?: string
+  config?: typeof defaultConfig
+}
+
+interface AppAuth extends Common {
+  client_id: string
+  client_secret?: string
+  refresh_token?: string
+  access_token?: string
+}
+
+interface ScriptAuth extends Common {
+  client_id: string
+  client_secret: string
+  username: string
+  password: string
+  two_factor_code?: number | string
+  access_token?: string
```

Comment:
Move types to separate files?

Reply:
@DedaDev Well, there are too many files in the project already, not sure if creating more is a good idea.